build/document-extractor.js

```diff
@@ -263,93 +287,166 @@ function _addSingleSectionBCD($) {
+            if (browserReleaseData.has(browser)) {
+              if (browserReleaseData.get(browser).has(added)) {
+                infoEntry.release_date = browserReleaseData
+                  .get(browser)
+                  .get(added).release_date;
+              }
+            }
+          }
+        }
+      }
+    }
+
     return [
       {
         type: "browser_compatibility",
         value: {
           title,
           id,
           isH3,
-          data: null,
+          data,
           query,
-          browsers: null,
+          browsers,
         },
       },
     ];
   }
 
-  // First extract a map of all release data, keyed by (normalized) browser
-  // name and the versions.
-  // You'll have a map that looks like this:
-  //
-  //   'chrome_android': {
-  //      '28': {
-  //        release_data: '2012-06-01',
-  //        release_notes: '...',
-  //        ...
-  //
-  // The reason we extract this to a locally scoped map, is so we can
-  // use it to augment the `__compat` blocks for the latest version
-  // when (if known) it was added.
-  const browserReleaseData = new Map();
-  for (const [name, browser] of Object.entries(browsers)) {
-    const releaseData = new Map();
-    for (const [version, data] of Object.entries(browser.releases || [])) {
-      if (data) {
-        releaseData.set(version, data);
-      }
-    }
-    browserReleaseData.set(name, releaseData);
-  }
+  function _buildSpecialSpecSection() {
+    // Collect spec_urls from a BCD feature.
+    // Can either be a string or an array of strings.
+    let specURLs = [];
 
-  for (const [key, compat] of Object.entries(data)) {
-    let block;
-    if (key === "__compat") {
-      block = compat;
-    } else if (compat.__compat) {
-      block = compat.__compat;
-    }
-    if (block) {
-      for (let [browser, info] of Object.entries(block.support)) {
-        // `info` here will be one of the following:
-        //  - a single simple_support_statement:
-        //    { version_added: 42 }
-        //  - an array of simple_support_statements:
-        //    [ { version_added: 42 }, { prefix: '-moz', version_added: 35 } ]
-        //
-        // Standardize the first version to an array of one, so we don't have
-        // to deal with two different forms below
-        if (!Array.isArray(info)) {
-          info = [info];
-        }
-        for (const infoEntry of info) {
-          const added = infoEntry.version_added;
-          if (browserReleaseData.has(browser)) {
-            if (browserReleaseData.get(browser).has(added)) {
-              infoEntry.release_date = browserReleaseData
-                .get(browser)
-                .get(added).release_date;
-            }
-          }
+    for (const [key, compat] of Object.entries(data)) {
+      if (key === "__compat" && compat.spec_url) {
+        if (Array.isArray(compat.spec_url)) {
+          specURLs = compat.spec_url;
+        } else {
+          specURLs.push(compat.spec_url);
```

Comment:
This algorithm is quite different from how we treat things in `_buildSpecialBCDSection` on line 354-359. 
In fact, I ran a measurement of ALL BCD queries possible and calculated how many times you get some `specURLs` if you just look in in `compat.spec_url` vs. possibly digging deeper into `compat.__compat.spec_url`:
```
{ hasSpecURLs: 3067, hasSpecURLs2: 3103, hasBCD: 8085, hasNotBCD: 262 }
```
(where `hasSpecURLs2` is an alternative implementation that checks if `compat.__compat` is the object to look inside)

Reply:
What's confusing is that sometimes there's something in `compat.spec_url` and sometimes it's in `compat.__compat.spec_url` and but here's a breakdown of all the BCD queries where the number of found `specURLs`s is different: https://gist.github.com/peterbe/f6e132913cf4cc185cd9aa1b6a5460bc depending on how you look. 

PS. I can't even remember why the BCD table does it like this. Sorry. Just seems a bit weird that the algorithm is different for `.support` vs. `.spec_url`.

Reply:
Great catch! I believe this only matters when BCD grouped things under a tree and that tree itself has no __compat (which is very rare).

The usual case (at least looking at your gist, this almost all of the cases, it is okay to not bother about what is below a tree, because the tree we've pointed at, e.g. `api.AbortController` has `api.AbortController.__compat` and that's exactly what we need for our specification link (BCD tables want more, they want all the sub features, too).

The edge case this deals with, though, is this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar with the BCD query: `javascript.grammar` which itself has no `javascript.grammar.__compat`. The BCD table will find something anyways though, as it looks into `javascript.grammar.whatever.__compat` and generate a table with all the features grouped as "grammar". 
I need to think about what we want here in terms of specs. In theory all these features have spec urls but that would obviously make a table as long as the compat table. Maybe this is actually a sign that the BCD and content organization is nuts and we should rather have separate pages for these things. Oh well :)

Reply:
So, I played with this and it is by design that I didn't go into the `compat.__compat.spec_urls`. As said in the earlier comment, unlike the BCD tables, we don't want to list all specs for all the sub features. So not this:

![Bildschirmfoto 2021-04-29 um 13 50 07](https://user-images.githubusercontent.com/349114/116546544-05ee7e00-a8f2-11eb-8c4e-32fea8fab7dd.png)

But rather just the main features' spec:

![Bildschirmfoto 2021-04-29 um 13 54 26](https://user-images.githubusercontent.com/349114/116546883-785f5e00-a8f2-11eb-81d6-2c71f61bd47f.png)

Reply:
And on the grammar page, we will need to see. Probably also not that useful to list specs for all the features that that compat table has.